apis/controller/v1alpha1/devworkspaceoperatorconfig_types.go

```diff
@@ -118,10 +118,11 @@ type WorkspaceConfig struct {
 	// "15m", "20s", "1h30m", etc. If not specified, the default value of "5m" is used.
 	ProgressTimeout string `json:"progressTimeout,omitempty"`
 	// IgnoredUnrecoverableEvents defines a list of Kubernetes event names that should
-	// be ignored when deciding to fail a DevWorkspace startup. This option should be used
-	// if a transient cluster issue is triggering false-positives (for example, if
-	// the cluster occasionally encounters FailedScheduling events). Events listed
-	// here will not trigger DevWorkspace failures.
+	// be ignored when deciding to fail a DevWorkspace startup.
+	// For example, a FailedScheduling event, that occurs when workspace cannot start
+	// due to exceeding available resources, should not fail the workspace startup, if there is
+	// an autoscaler configured on the cluster, and we want to wait until it provisions additional resources.
+	// FailedScheduling event can also occur as a false-positive, as a result of a transient cluster issue.
```

Comment:
I suggest experimenting with [kubebuilder annotations](https://book.kubebuilder.io/reference/markers/crd-validation) for the IgnoredUnrecoverableEvents field. 

We should try setting the default array value. I think this would be done with `+kubebuilder:default:={"FailedScheduling"}`

I believe that should be enough to populate the IgnoredUnrecoverableEvents list in the DWOC. Make sure you re-generate the CRD's in a seperate commit by running: `make update_devworkspace_api update_devworkspace_crds generate_all`

Something to note: This entire PR might be dropped and re-implemented in Che-Operator if we can get the kubebuilder approach working. We'd want Che admins to _see_ that the FailedSchedling event is ignored by default & there would be no advantages to duplicating this code change in both DWO & Che-Operator (unless users who use DWO in isolation want this feature, however, this is not the current reason why we're resolving https://github.com/devfile/devworkspace-operator/issues/1280).